docs/website/scripts/generate_header.py

```diff
@@ -0,0 +1,46 @@
+import tomllib
+from typing import NamedTuple
+
+
+class NavigationItem(NamedTuple):
+    name: str
+    url: str
+    icon: str
+    icon_only: bool = False
+
+
+def generate_navigation_for(items: list[NavigationItem], mobile: bool = False) -> str:
+    # TODO: Add outlink icon
+    item_template = "<a href=\"URL\" class=\"nav-link nav-icon\">NAME_OR_ICON</a>"
```

Comment:
Use string formatting with variable names to make this less prone to errors if changed (`replace` doesn't raise an error if the term doesn't exist in the string)